From 0db9c35d408a3a2d56704ac4a70d5fd3ebda706f Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Tue, 2 May 2017 13:47:08 +0100
Subject: [PATCH 9/9] cirrus: fix PUTPIXEL macro

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <20170502134708.21817-6-kraxel@redhat.com>
Patchwork-id: 74989
O-Subject: [RHEL-6.9 qemu-kvm PATCH 5/5] cirrus: fix PUTPIXEL macro
Bugzilla: 1444378 1444380
RH-Acked-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Thomas Huth <thuth@redhat.com>

Should be "c" not "col".  The macro is used with "col" as third parameter
everywhere, so this tyops doesn't break something.

Fixes: 026aeffcb4752054830ba203020ed6eb05bcaba8
Reported-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
Reviewed-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
Message-id: 1490168303-24588-1-git-send-email-kraxel@redhat.com
(cherry picked from commit db6cd4c855fd7a9e397bafd3361e66509fe47de0)
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
---
 hw/cirrus_vga_rop2.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hw/cirrus_vga_rop2.h b/hw/cirrus_vga_rop2.h
index b86bcd6..b208b73 100644
--- a/hw/cirrus_vga_rop2.h
+++ b/hw/cirrus_vga_rop2.h
@@ -29,8 +29,8 @@
 #elif DEPTH == 24
 #define PUTPIXEL(s, a, c)    do {          \
         ROP_OP(s, a,     c);               \
-        ROP_OP(s, a + 1, (col >> 8));      \
-        ROP_OP(s, a + 2, (col >> 16));     \
+        ROP_OP(s, a + 1, (c >> 8));        \
+        ROP_OP(s, a + 2, (c >> 16));       \
     } while (0)
 #elif DEPTH == 32
 #define PUTPIXEL(s, a, c)    ROP_OP_32(s, a, c)
-- 
1.8.3.1

